lib/media/text_engine.js

```diff
@@ -110,7 +110,47 @@ shaka.media.TextEngine.makeCue = function(startTime, endTime, payload) {
     return null;
   }
 
-  return new VTTCue(startTime, endTime, payload);
+  if (window.VTTCue &&
+      (/\[native code\]/).test(window.VTTCue.toString())) {
+    return new window.VTTCue(startTime, endTime, payload);
+  }
+
+  if (!window.TextTrackCue) {
+    shaka.log.error('VTTCue not available.');
+    return null;
+  }
+
+  var canUse3ArgsTextTrackCue = function() {
+    try {
+      return !!new window.TextTrackCue(1, 2, '');
+    } catch (error) {
+      return false;
+    }
+  };
+
+  var constructorLength = window.TextTrackCue.length;
+
+  // Draft spec TextTrackCue with 3 constructor arguments.
+  // See {@link https://goo.gl/ZXBWZi W3C Working Draft 25 October 2012}.
+  if (constructorLength == 3) {
+    return new window.TextTrackCue(startTime, endTime, payload);
+
+  // Draft spec TextTrackCue with 6 constructor arguments (5th & 6th are
+  // optional).
+  // See {@link https://goo.gl/AYFqUh W3C Working Draft 29 March 2012}.
+  // Quoting the access to the TextTrackCue object to avoid the compiler
+  // complaining.
+  } else if (constructorLength == 6) {
+    var id = startTime + '-' + endTime + '-' + payload;
+    return new window['TextTrackCue'](id, startTime, endTime, payload);
```

Comment:
It's from the original polyfill https://github.com/google/shaka-player/blob/master/lib/polyfill/vttcue.js#L91, Looking through the history I don't see a reason for why this is using brackets.